### PR TITLE
Some additional clean-ups

### PR DESCRIPTION
**Pull request by colesbury**
- Remove some uses of mega-header THP.h
 - Use HANDLE_TH_ERRORS in functions that may throw
 - Move NumPy includes to common header
 - Delete unused allocator

### CODE REVIEW

**Comment by soumith** [comment]
onnx tests failing:

```
torch/csrc/Module.cpp:521:3: error: 'END_HANDLE_TH_ERRORS' was not declared in this scope
11:08:51        END_HANDLE_TH_ERRORS
```

File: torch/csrc/Module.cpp [hunk 1]
```diff
@@ -442,7 +443,7 @@ static PyObject* initModule() {
   ASSERT_TRUE(THPWrapper_init(module));
   ASSERT_TRUE(THPGenerator_init(module));
   ASSERT_TRUE(THPException_init(module));
-  ASSERT_TRUE(THPSize_init(module));
+  THPSize_init(module);
```
Comment (gchanan) [inline torch/csrc/Module.cpp]: Why'd you change this?  Seems nicer to just have everything return a boolean for consistency, even if they can't be false.

File: torch/csrc/Size.cpp [hunk 2]
```diff
@@ -39,6 +34,7 @@ static PyObject * THPSize_pynew(PyTypeObject *type, PyObject *args, PyObject *kw
 
 static PyObject * THPSize_repr(THPSize *self)
 {
+  HANDLE_TH_ERRORS
```
Comment (gchanan) [inline torch/csrc/Size.cpp]: can we rename this macro?  I thought I only needed to guard things when I call into TH, but here we never touch TH.

**Review by gchanan — APPROVED** [review]

File: torch/csrc/Module.cpp (hunk 1 above)
Comment (colesbury) [reply torch/csrc/Module.cpp]: Nothing should return a boolean, but I'm too lazy to clean-up everything at once.

The whole `ASSERT_TRUE` thing ends up replacing useful error messages with worthless ones.

File: torch/csrc/Size.cpp (hunk 2 above)
Comment (colesbury) [reply torch/csrc/Size.cpp]: Yeah, that seems like a good idea. (We need the macro around any calls which may throw C++ exceptions)

**Pull request merged**